osfclient/cli.py

```diff
@@ -66,6 +69,33 @@ def _setup_osf(args):
     return OSF(username=username, password=password)
 
 
+def init(args):
+    """Initialize or edit an existing .osfcli.config file
+
+    """
+    # reading existing config file, convert to configparser object
+    config_ = configparser.ConfigParser()
+    config_.read_dict({'osf': config_from_file()})
+
+    print('Provide a username for the config file '
+          '[current username: {}]:'.
+          format(config_.get('osf', 'username', fallback='')))
```

Comment:
instead of printing this you should be able to give it as argument to `input`, and then in the test we can assert that vaguely the right questions were asked (without also having to learn how to capture the stdout)